x-pack/elastic-agent/pkg/agent/operation/operator.go

```diff
@@ -218,9 +218,20 @@ func (o *Operator) Shutdown() {
 		o.logger.Debugf("pipeline installer '%s' done", o.pipelineID)
 	}
 
-	for _, app := range o.apps {
-		app.Shutdown()
+	wg := sync.WaitGroup{}
+	started := time.Now()
+	for _, a := range o.apps {
+		// shutdown apps concurrently.
+		// TODO(Anderson): it's fine, right?
+		wg.Add(1)
+		go func(a Application) {
+			a.Shutdown()
+			wg.Done()
+		}(a)
```

Comment:
I don't think shutting it down in parallel would be a problem, we were still doing a blocking call before.
Instead of `wg.add()` in each iteration you can get the size on the map `len(o.apps)`. Looking at the code, there is a lock for the manipulation on the `apps`, we might want to acquire the lock and release it after shutdown, so the value in the map is fixed. I believe the when we are calling `shutdown` the map should stay consistent but  still it might better to do it.